src/common/security/admiral/context.go

```diff
@@ -71,19 +75,20 @@ func (s *SecurityContext) IsSolutionUser() bool {
 
 // Can returns whether the user can do action on resource
 func (s *SecurityContext) Can(action rbac.Action, resource rbac.Resource) bool {
-	ns, err := resource.GetNamespace()
-	if err == nil {
-		switch ns.Kind() {
-		case "project":
+	s.once.Do(func() {
+		s.evaluator = rbac.NewNamespaceEvaluator("project", func(ns rbac.Namespace) rbac.Evaluator {
 			projectID := ns.Identity().(int64)
-			isPublicProject, _ := s.pm.IsPublic(projectID)
-			projectNamespace := rbac.NewProjectNamespace(projectID, isPublicProject)
-			user := project.NewUser(s, projectNamespace, s.GetProjectRoles(projectID)...)
-			return rbac.HasPermission(user, resource, action)
-		}
-	}
+			proj, err := s.pm.Get(projectID)
+			if err != nil || proj == nil {
```

Comment:
Let's explicitly log the error, b/c it seems all the subsequent `Can` will return false.

Reply:
Fixed, the error message was send to log when failed to get project info.